Pull request

igoreskin opened the pull request:
*Please check off the stories that have been completed in your homework submission. Feel free to add any additional notes to the PR description.*


### User stories to implement
 - [x] A user should not be able to complete a task when creating it.
 - [x] A user can filter their view by complete or incomplete tasks.
 - [x] A user's default view is filtered to show only incomplete tasks.
 - [ ] A user should be able to keep multiple lists.
 - [x] A user should have access to only their tasks. (may require devise or a creative, lightweight auth system)
 - [ ] A user viewing a task should see a history of all its changes. (task audit trail)
 - [ ] A user should be able to see a history of all completed tasks and associated actions. (app audit trail)

### Cleanup tasks
- [ ] Consolidate the repeated code between the create and edit feature specs for tasks.


### Review Status
**Note to candidate:** Do not edit this section

- [ ] Specs Passing
- [ ] Two approving reviews

sethwebster commented:
@igoreskin Can you resolve the failing build? We will review once the build is passing.

Thanks!

igoreskin commented:
Hi Seth!

I'll do it tonight immediately after work.

Thanks,

Igor

On Mon, 15 Oct 2018 at 10:22, Seth Webster <notifications@github.com> wrote:

> @igoreskin <https://github.com/igoreskin> Can you resolve the failing
> build? We will review once the build is passing.
>
> Thanks!
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/contently/todo/pull/28#issuecomment-429872460>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AZMZNycW5yi1HPCME99A8dwizd_NHomHks5ulJoagaJpZM4XbdVn>
> .
>

sethwebster commented:
@igoreskin I have fixed the submission for you. This required following the instructions:

`You are trying to install in deployment mode after changing
your Gemfile. Run `bundle install` elsewhere and add the
updated Gemfile.lock to version control.

You have deleted from the Gemfile:
* wdm (>= 0.1.0)`

I simply downloaded your code, removed the Gemfile.lock, ran `bundle install` and committed the new Gemfile.lock.

sethwebster commented:
@igoreskin The app is fixed, from a build perspective, however, the specs are failing.  If you wish to proceed, please fix those failing specs. We will review once we have a passing build.

Thanks!

igoreskin commented:
Thanks an awful lot! I tried to look for instructions everywhere and I ran
all the commands I'd found in GitHub, but it didn't occur to me to remove
the Gemflie.loc - although that's exactly what you do in Node.js, you just
remove the packge-lock.sjon and re-run npm install. I definitely should've
thought of that.

Thanks again,

Igor

On Tue, 16 Oct 2018 at 09:35, Seth Webster <notifications@github.com> wrote:

> @igoreskin <https://github.com/igoreskin> I have fixed the submission for
> you. This required following the instructions:
>
> You are trying to install in deployment mode after changing your Gemfile.
> Runbundle install` elsewhere and add the
> updated Gemfile.lock to version control.
>
> You have deleted from the Gemfile:
>
>    - wdm (>= 0.1.0)`
>
> I simply downloaded your code, removed the Gemfile.lock, ran bundle
> install and committed the new Gemfile.lock.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/contently/todo/pull/28#issuecomment-430239668>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AZMZN-CDwI8aB5olQByctkLGzDdP0FZYks5uleCdgaJpZM4XbdVn>
> .
>

jamescontent commented on app/controllers/tasks_controller.rb:
```diff
@@ -6,12 +6,19 @@ class TasksController < ApplicationController
   # GET /tasks.json
   def index
-    @tasks = Task.all
+    @completion = Task.completion 
```
Is there a more descriptive name we could use for this variable? Or what it is used for?

finleye commented on app/models/task.rb:
```diff
@@ -2,4 +2,21 @@
 
 class Task < ActiveRecord::Base
```
Should this class do any validations to ensure that no instances are created with `completed: true`?

igoreskin commented:
Hi Corey and James!

Thanks a lot for your emails, sorry about getting back so late - my workday
ended not long ago. The question about validation in the task class -
you're right, I actually thought about it, but then decided that it would
be kind of redundant, because the checkbox is not even displayed to the
user at the moment of creation because the task doesn't have a name yet,
i.e. it has not been created previously, so user can only check the box
when updating the task. In the partial views/tasks/_form.html.erb:

<% if @task.name %>
<div class="field">
<%= f.label :completed %>
<%= f.check_box :completed %>
</div>
<% end %>

They syntax for validation of a Boolean would probably be
validates_acceptance_of :completed, accept: false, or validates :completed,
inclusion: {:in => [false]}, although this second option is kind of clumsy.
There may actually be a better syntax.

The name of the variable completion in the tasks controller is indeed very
non-descriptive, much better would be 'status' or maybe
'completion_status',although the latter may be too long.

The very first question about passing attributes through in devise - I may
have misunderstood the question, but your links 'docs' and 'something like
this' point to the exact same page in documentation, could you please
clarify?

Concerning more succinct way to express what I wrote in those two class
methods in the task model - I took the syntax directly from my rails
project where I selected fashion shows by location, and I remember thinking
long and carefully how to do it, and that was the best way I could come up
with, besides it allows to keep the controller action (index) very clean
and readable. To be honest I don't know how else to express it - I
certainly need both methods in the controller to enable the select tag with
options_for_select. I would actually appreciate a hint on how else it could
be done.

Thanks a lot again and best regards

Igor

On Tue, 16 Oct 2018 at 15:10, Corey Finley <notifications@github.com> wrote:

> *@finleye* commented on this pull request.
> ------------------------------
>
> In app/models/task.rb
> <https://github.com/contently/todo/pull/28#discussion_r225671200>:
>
> > @@ -2,4 +2,21 @@
>
>  class Task < ActiveRecord::Base
>
> Should this class do any validations to ensure that no instances are
> created with completed: true?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/contently/todo/pull/28#pullrequestreview-165320947>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AZMZN-I-OLi-VUk1weohe1YV_3jQFwNNks5uli8LgaJpZM4XbdVn>
> .
>
